File: Data Structures & Algorithms/duplicate-integer/submission-1.java
public class Solution{
    public static void main(String[] args){
        int[] nums = {1,2,3,4};
        
        System.out.print(hasDuplicate(nums));
    }

    public static boolean hasDuplicate(int[] nums){
        for(int i = 0; i < nums.length-1; i++){
            for(int j = i + 1; j< nums.length; j++){
                if(nums[i] == nums[j]){
                    return true;
                }
                
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/contains-duplicate-ii/submission-1.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        # window = set()
        # L = 0
        # for R in range(len(nums)):
        #     if R - L > k:
        #         window.remove(nums[L])
        #         L += 1
        #     if nums[R] in window:
        #         return True
        #     window.add(nums[R])
        # return False

        #hashmap
        seen = {}
        for i, n in enumerate(nums):
            if nums[i] in seen and i - seen[n] <= k:
                return True
            seen[n] = i
        return False
                    